src/utilities.js

```diff
@@ -250,3 +251,92 @@ export function getResults(request) {
     });
   };
 }
+
+export function getTrendingResults(request) {
+  return (searchType, obj) => {
+    const map = {
+      'Daily trends': {
+        path: '/trends/api/dailytrends',
+        extraParams: {
+          ed: convertDateToString(obj.trendDate, false, true),
+          ns: obj.ns,
+        },
+      },
+      'Real time trends': {
+        path: '/trends/api/realtimetrends',
+        extraParams: {
+          fi: 0,
+          fs: 0,
+          ri: 300, // # of trending stories IDs returned
+          rs: 20,
+          sort: 0,
+        },
+      },
+    };
+
+    let options = {
+      method: 'GET',
+      host: 'trends.google.com',
+      path: map[searchType].path,
+      qs: {
+        hl: obj.hl,
+        tz: obj.timezone,
+        geo: obj.geo,
+        cat: obj.category,
+      },
+    };
+
+    if (obj.agent) options.agent = obj.agent;
+
+    Object.assign(options.qs, options.qs, map[searchType].extraParams);
+
+    return request(options)
+    .then((res) => {
+      try {
+        /** JSON.parse will decode unicode */
+        const results = JSON.stringify(JSON.parse(res.slice(5)));
+
+        return results;
+      } catch (e) {
+        /** throws if not valid JSON, so just return unaltered res string */
+        return res;
+      }
+    });
+  };
+}
+
+export function constructTrendingObj(obj, cbFunc) {
+  if (typeof obj === 'function') cbFunc = obj;
+
+  if (!obj || !!obj && typeof obj !== 'object' || Array.isArray(obj)) {
```

Comment:
Can you take advantage of the `validateObj` function in `utilities.js` or generalize that function so it will work for both cases? Seems like we are doing a lot of the same work twice

Reply:
So, I tried to reuse the constructObj function, re writing it to make it aware of the differences between the interest type searches and trending searches and the code became incredibly ugly. It appears like there could be a lot of code reuse but in reality there are differences if you look closely. I could break out the similar parts more, but it just didn't seem worth it in the end.